### PR TITLE
ES2015 Rest parameters sample

### PR DESCRIPTION
**Pull request by addyosmani**
r: @jeffposnick

### CODE REVIEW

File: rest-parameters-es6/index.html [hunk 1]
```diff
@@ -0,0 +1,83 @@
+---
+feature_name: Rest parameters (ES2015)
+chrome_version: 47
+feature_id: 5937087912083456
+---
+
+<h3>Background</h3>
+<p>
+Rest parameters allow your functions to have variable number of arguments without using the arguments object. The rest parameter is an instance of Array so all the array methods just work.
```
Comment (jeffposnick) [inline rest-parameters-es6/index.html]: Rest parameters → [Rest parameters](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters)

arguments → [`arguments`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments)

File: rest-parameters-es6/index.html [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+</p>
```
Comment (jeffposnick) [inline rest-parameters-es6/index.html:10]: Can you mention in the Background section that there can only be a single rest parameter for a given function? It's obvious when you know about the feature, but I could imagine some folks newer folks wondering what happens if you try `function foo(...first, ...second)`

File: rest-parameters-es6/index.html [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  animals.forEach(function(animal) {
+    ChromeSamples.log(animal);
+  });
+}
+
+listAnimalsES2015('🐯', '🐰', '🐘', '🐴', '🐥');
+
+// This approach becomes even simpler with ES2015 fat arrows
+function listAnimalsArrows(...animals) {
+  animals.forEach((animal) => ChromeSamples.log(animal));
```
Comment (jeffposnick) [inline rest-parameters-es6/index.html]: (animal) → animal

File: rest-parameters-es6/index.html [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+listAnimalsArrows('🐯', '🐰', '🐘', '🐴', '🐥');
+
+// Get the count of elements using the length property
+function fun1(...theArgs) {
```
Comment (jeffposnick) [inline rest-parameters-es6/index.html]: fun1 → countArguments (or something else that's more descriptive)

**Comment by jeffposnick** [comment]
Reviewed. Thanks!

File: rest-parameters-es6/index.html [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+ChromeSamples.log(sortRestArgs(5,3,7,1)); // shows 1,3,5,7
+
+function sortArguments() {
+  var sortedArgs = arguments.sort();
+  return sortedArgs; // this will never happen
+}
+
+// throws a TypeError: arguments.sort is not a function
```
Comment (beaufortfrancois) [inline rest-parameters-es6/index.html:82]: To be fair here, we could mention `Array.from`:

``` js
function sortArguments() { 
  return Array.from(arguments).sort(); 
};
```

**Comment by addyosmani** [comment]
Thanks for reviewing. Attempted to address comments. PTAL.

File: rest-parameters-es6/index.html [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+chrome_version: 47
+feature_id: 5937087912083456
+---
+
+<h3>Background</h3>
+<p>
+<a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters">Rest parameters</a> allow your functions to have variable number of arguments without using the <a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments">arguments</a> object. The rest parameter is an instance of Array so all the array methods just work.
+</p>
+
+<p>Note that there can only be a single rest parameter for a given function. This means that `listAnimals(...cats, ...dogs)` will not work, whilst `listAnimals(...cats)` will work just fine.</p>
```
Comment (jeffposnick) [inline rest-parameters-es6/index.html]: The backtick escapes need to be `<code>` here, since it's just plain HTML.

**Comment by jeffposnick** [comment]
LGTM after the change to `<code>` is made.

**Comment by addyosmani** [comment]
Correction made. Thanks for reviewing!

**Pull request merged**